File: app/src/main/java/com/bq/zowi/interactors/ChangeZowiNameInteractorImpl.java
package com.bq.zowi.interactors;

import com.bq.zowi.models.ZowiName;
import com.bq.zowi.models.commands.NameSetCommand;
import rx.Single;

/* JADX INFO: loaded from: classes.dex */
public class ChangeZowiNameInteractorImpl implements ChangeZowiNameInteractor {
    private final SendCommandToZowiInteractor sendCommandToZowiInteractor;

    public ChangeZowiNameInteractorImpl(SendCommandToZowiInteractor sendCommandToZowiInteractor) {
        this.sendCommandToZowiInteractor = sendCommandToZowiInteractor;
    }

    @Override // com.bq.zowi.interactors.ChangeZowiNameInteractor
    public Single<Void> changeZowiName(String name) {
        return this.sendCommandToZowiInteractor.sendCommandToZowi(new NameSetCommand(name));
    }

    @Override // com.bq.zowi.interactors.ChangeZowiNameInteractor
    public Single<Void> resetZowiNameToFactory() {
        return this.sendCommandToZowiInteractor.sendCommandToZowi(new NameSetCommand(ZowiName.getFactoryName()));
    }
}

File: zowi-core/src/main/kotlin/com/bq/zowi/usecases/ChangeZowiNameInteractorImpl.kt
package com.bq.zowi.usecases

import com.bq.zowi.models.ZowiName
import com.bq.zowi.models.commands.NameSetCommand
import io.reactivex.Completable

class ChangeZowiNameInteractorImpl(
    private val sendCommandToZowiInteractor: SendCommandToZowiInteractor
) : ChangeZowiNameInteractor {

    override fun changeZowiName(name: String): Completable {
        return sendCommandToZowiInteractor.sendCommandToZowi(NameSetCommand(name))
    }

    override fun resetZowiNameToFactory(): Completable {
        return sendCommandToZowiInteractor.sendCommandToZowi(NameSetCommand(ZowiName.getFactoryName()))
    }
}
